azure-kit-starter/src/main/java/com/vaadin/azure/starter/ui/ClusterSupport.java

```diff
@@ -0,0 +1,174 @@
+import com.vaadin.flow.internal.CurrentInstance;
+import com.vaadin.flow.server.Command;
+import com.vaadin.flow.server.RequestHandler;
+import com.vaadin.flow.server.ServiceInitEvent;
+import com.vaadin.flow.server.VaadinRequest;
+import com.vaadin.flow.server.VaadinResponse;
+import com.vaadin.flow.server.VaadinServiceInitListener;
+import com.vaadin.flow.server.VaadinSession;
+
+/**
+ * Cluster support for Vaadin applications. This component allows receiving
+ * events from the cluster running the Vaadin application.
+ */
+public class ClusterSupport implements VaadinServiceInitListener {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Version environment variable name.
+     */
+    public static final String ENV_APP_VERSION = "APP_VERSION";
+
+    /**
+     * Current version cookie name.
+     */
+    public static final String CURRENT_VERSION_COOKIE = "app-version";
+
+    /**
+     * Update version cookie name.
+     */
+    public static final String UPDATE_VERSION_COOKIE = "app-update";
+
+    /**
+     * Sticky cluster cookie name.
+     */
+    public static final String STICKY_CLUSTER_COOKIE = "INGRESSCOOKIE";
+
+    private SwitchVersionListener switchVersionListener;
+
+    private String appVersion;
+
+    /**
+     * Get the current instance of the ClusterSupport.
+     */
+    public static ClusterSupport getCurrent() {
+        return CurrentInstance.get(ClusterSupport.class);
+    }
+
+    /**
+     * Register the global version switch listener. If set to <code>null</code>
+     * the current session and the cookies are removed without any version
+     * switch condition check.
+     *
+     * @param listener
+     *            the listener to register.
+     */
+    public void setSwitchVersionListener(SwitchVersionListener listener) {
+        this.switchVersionListener = listener;
+    }
+
+    @Override
+    public void serviceInit(ServiceInitEvent serviceInitEvent) {
+        appVersion = System.getenv(ENV_APP_VERSION);
+        if (appVersion == null) {
+            getLogger().error(
+                    "Missing environment variable 'APP_VERSION'. ClusterSupport service not initialized.");
+            return;
+        }
+        getLogger().info(
+                "ClusterSupport service initialized. Registering RequestHandler with Application Version: "
+                        + appVersion);
+
+        // Register a generic request handler for all the requests
+        serviceInitEvent.addRequestHandler(createRequestHandler());
+    }
+
+    private RequestHandler createRequestHandler() {
+        return (vaadinSession, vaadinRequest, vaadinResponse) -> {
+            // Set the thread local instance
+            CurrentInstance.set(ClusterSupport.class, this);
+
+            vaadinSession.access(handleVersions(vaadinSession, vaadinResponse));
+
+            // If the current and the new versions do not match notify the user
+            return false;
+        };
+    }
```

Comment:
We can further improve readability by refactoring the method to be the `RequestHandler` itself, and use a method reference when adding it to the service event, i.e.

```suggestion
        // Register a generic request handler for all the requests
        serviceInitEvent.addRequestHandler(this::handleRequest);
    }

    private boolean handleRequest(VaadinSession vaadinSession,
            VaadinRequest vaadinRequest, VaadinResponse vaadinResponse) {
        // Set the thread local instance
        CurrentInstance.set(ClusterSupport.class, this);

        vaadinSession.access(handleVersions(vaadinSession, vaadinResponse));

        // If the current and the new versions do not match notify the user
        return false;
    }
```

Reply:
Refactored to use method reference and moved the command part back to inline.